pkg/declarativeconfig/transform/auth_provider.go

```diff
@@ -0,0 +1,151 @@
+package transform
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/authproviders/iap"
+	"github.com/stackrox/rox/pkg/auth/authproviders/oidc"
+	"github.com/stackrox/rox/pkg/auth/authproviders/openshift"
+	"github.com/stackrox/rox/pkg/auth/authproviders/saml"
+	"github.com/stackrox/rox/pkg/auth/authproviders/userpki"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+var _ Transformer = (*authProviderTransform)(nil)
+
+type authProviderTransform struct{}
+
+func newAuthProviderTransformer() *authProviderTransform {
+	return &authProviderTransform{}
+}
+
+func (a *authProviderTransform) Transform(configuration declarativeconfig.Configuration) (map[reflect.Type][]proto.Message, error) {
+	authProviderConfig, ok := configuration.(*declarativeconfig.AuthProvider)
+	if !ok {
+		return nil, errox.InvalidArgs.Newf("invalid configuration type received for auth provider: %T", configuration)
+	}
+	authProviderProto := &storage.AuthProvider{
+		Id:                 declarativeconfig.NewDeclarativeAuthProviderUUID(authProviderConfig.Name).String(),
+		Name:               authProviderConfig.Name,
+		Type:               getType(authProviderConfig),
+		UiEndpoint:         authProviderConfig.UIEndpoint,
+		Enabled:            true,
+		Config:             getConfig(authProviderConfig),
+		LoginUrl:           "/sso/login/" + declarativeconfig.NewDeclarativeAuthProviderUUID(authProviderConfig.Name).String(),
+		Active:             true,
```

Comment:
So, `Active==true` when the first user logged in, and AFAICT will never be reset. This will lead to auth provider editing being disabled. I chose to set it to true, since it doesn't really make a difference for the user since they won't be able to modify the auth provider either way.

`Enabled==true` means that the auth provider will be visible as a login auth provider, i.e. the auth provider can be used during login in the drop down.

I added some comments explaining this, I think based on how we want to store the providers, we might decide to remove them in the future and rely on the options, similar to the login URL comment you had.